scipy/optimize/minpack.py

```diff
@@ -449,23 +448,37 @@ def leastsq(func, x0, args=(), Dfun=None, full_output=0,
         return (retval[0], info)
 
 
-def _wrap_func(func, xdata, ydata, weights):
-    if weights is None:
+def _wrap_func(func, xdata, ydata, T):
```

Comment:
It seems to me that, depending on the value of parameter `T` (quite mysterious at this point by the way), the function does a rather different job. Same for `_wrap_jac` below. So I wonder if there shouldn't be two distinct functions instead.